src/SArray.jl

```diff
@@ -239,6 +252,69 @@ function static_array_gen(::Type{SA}, @nospecialize(ex), mod::Module) where {SA}
                         )
                     end
                 end
+            elseif _isnonnegvec(ex.args[4:end])
+                # for calls like `rand(rng, type, dims...)`
+                # for calls like `rand(rng, sampler, dims...)`
+                return quote
+                    StaticArrays._rand(
+                        $(esc(ex.args[2])),
+                        $(esc(ex.args[3])),
+                        Size($(escall(ex.args[4:end])...)),
+                        $SA{
+                            Tuple{$(escall(ex.args[4:end])...)},
+                            Random.gentype($(esc(ex.args[3]))),
+                        },
+                    )
+                end
+            else
+                error("@$SA got bad expression: $(ex)")
+            end
+        elseif f === :randn || f === :randexp
+            _f = Symbol(:_, f)
+            if length(ex.args) == 1
+                # No support for `@SArray randn()` etc.
+                error("@$SA got bad expression: $(ex)")
+            elseif _isnonnegvec(ex.args[2:end])
+                # for calls like `randn(dims...)`
+                return :($f($SA{$Tuple{$(escall(ex.args[2:end])...)}}))
+            elseif _isnonnegvec(ex.args[3:end])
```

Comment:
```suggestion
            elseif length(ex.args) >= 3 && _isnonnegvec(ex.args[3:end])
```
To avoid indexing out of range.

Reply:
If `length(ex.args) < 3`, then `3:end` will be empty, and the function `_isnonnegvec` returns `false` for empty vectors.

```julia
julia> import StaticArrays._isnonnegvec

julia> args = [1,2,3]
3-element Vector{Int64}:
 1
 2
 3

julia> _isnonnegvec(args[5:end])
false
```

Reply:
Right, I forgot about that feature of range indexing.